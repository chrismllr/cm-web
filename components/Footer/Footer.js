import React from 'react'
import MaterialIcon from '../MaterialIcon'
import format from 'date-fns/format'

const footerIcon = {
  marginRight: '.5rem',
  paddingLeft: '.1rem'
}

export default () => (
  <div className="Footer">
    <div className="container Footer-wrapper">
      <div className="Footer-col1">
        <h3 data-hook="questions" className="AnyQuestions">Any questions?</h3>
        <a href="mailto:chrsmllr@gmail.com" data-hook="questions-mailto">You got my digits.</a>
        <p data-hook="copyright">&copy; Chris Miller {format(new Date(), 'YYYY')}</p>
      </div>

      <div className="Footer-col2">
        <a><MaterialIcon icon="mail" style={footerIcon} /> E-mail</a>
        <a><MaterialIcon icon="instagram" style={footerIcon} /> Instagram</a>
        <a><MaterialIcon icon="facebook" style={footerIcon} /> Facebook</a>
      </div>
    </div>

    <style jsx global>{`
      .Footer {
        padding: 2rem 0;
      }

      .Footer-wrapper {
        display: flex;
      }

      .Footer-col1, .Footer-col2 {
        width: 50%;
      }

      .Footer-col1 {
        margin-right: 1.5rem;
      }

      .Footer-col1 p {
        margin-top: 2rem;
      }

      .Footer p, .Footer a {
        font-size: .83rem;
        display: block;
        opacity: 0.5;
      }

      .AnyQuestions {
        color: #222;
        margin-bottom: .5rem;
        line-height: 1.2;
      }

      .Footer a {
        transition: .5s ease-in-out;
        cursor: pointer;
      }

      .Footer a:hover {
        opacity: 1;
      }

      @media (min-width: 630px) {
        .Footer {
          padding: 4rem 0;
        }
      }
    `}</style>
  </div>
)
