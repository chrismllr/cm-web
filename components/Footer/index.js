import React from 'react'
import format from 'date-fns/format'
import { FlexContainer, Section } from '../Layout'

export default () => (
  <Section className="Footer">
    <FlexContainer>
      <div className="Footer-col1">
        <h3 data-hook="questions" className="AnyQuestions">Any questions?</h3>
        <a className="Digits-link" href="mailto:chrsmllr@gmail.com" data-hook="questions-mailto">You got my digits.</a>
        <p className="Footer-dim" data-hook="copyright">&copy; Chris Miller {format(new Date(), 'YYYY')}</p>
      </div>

      <div className="Footer-col2">
        <a className="Footer-dim" href="http://github.com/chrismllr" target="_blank" rel="noopener" data-hook="github">
          <span className="Arrow">></span> github
        </a>
        <a className="Footer-dim" href="mailto:chrsmllr@gmail.com">
          <span className="Arrow">></span> e-mail
        </a>
      </div>
    </FlexContainer>

    <style jsx>{`
      section {
        background-color: #f8f8f8;
      }

      .Footer-col1, .Footer-col2 {
        width: 50%;
      }

      .Footer-col1 {
        margin-right: 1.5rem;
      }

      .Footer-col1 p {
        margin-top: 2rem;
      }

      .Footer-dim {
        display: block;
        letter-spacing: .01rem;
        color: #6b6b6b;
        text-decoration: none;
      }

      .Digits-link {
        display: block;
        letter-spacing: .01rem;
        color: #FB0101;
        text-decoration: underline;
      }

      .Arrow {
        color: #222;
        font-weight: 700;
        font-size: 1.25rem;
        line-height: 1;
        margin-right: .2rem;
      }


      .AnyQuestions {
        color: #222;
        margin-bottom: .5rem;
        line-height: 1.2;
      }

      .Footer a {
        transition: .5s ease-in-out;
        cursor: pointer;
      }

      .Footer a:hover {
        opacity: 1;
      }

      @media (min-width: 630px) {
        .Footer {
          padding: 4rem 0;
        }
      }
    `}</style>
  </Section>
)
