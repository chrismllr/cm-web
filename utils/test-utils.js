export function hook (identifier) {
  return '[data-hook="' + identifier + '"]'
}
