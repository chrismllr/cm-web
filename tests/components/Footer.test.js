import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Footer from '../../components/Footer'
import { hook } from '../test-utils'
import format from 'date-fns/format'

describe('(Component) Footer', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Footer />
    )
  })

  it('Renders Any Questions with a mailto link below', () => {
    expect(wrapper.find(hook('questions')).text()).toEqual('Any questions?')
    expect(wrapper.find(hook('questions-mailto')).text()).toEqual('You got my digits.')
    expect(wrapper.find(hook('questions-mailto')).prop('href')).toEqual('mailto:chrsmllr@gmail.com')
  })

  it('Renders copyright with correct year', () => {
    expect(wrapper.find(hook('copyright')).text()).toEqual(`Â© Chris Miller ${format(new Date(), 'YYYY')}`)
  })

  it('Renders three MaterialIcon`s for social links', () => {
    expect(wrapper.find('MaterialIcon').length).toEqual(3)
  })

  it('Renders to snapshot', () => {
    const component = renderer.create(<Footer />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
