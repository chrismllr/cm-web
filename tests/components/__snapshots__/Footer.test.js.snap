// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Footer Renders to snapshot 1`] = `
<div
  className="Footer"
  data-jsx={3314000957}
>
  <div
    className="container Footer-wrapper"
    data-jsx={3314000957}
  >
    <div
      className="Footer-col1"
      data-jsx={3314000957}
    >
      <h3
        className="AnyQuestions"
        data-hook="questions"
        data-jsx={3314000957}
      >
        Any questions?
      </h3>
      <a
        data-hook="questions-mailto"
        data-jsx={3314000957}
        href="mailto:chrsmllr@gmail.com"
      >
        You got my digits.
      </a>
      <p
        data-hook="copyright"
        data-jsx={3314000957}
      >
        Â© Chris Miller 
        2017
      </p>
    </div>
    <div
      className="Footer-col2"
      data-jsx={3314000957}
    >
      <a
        data-jsx={3314000957}
      >
        <i
          className="icon icon-mail"
          data-jsx={1566661193}
          style={
            Object {
              "marginRight": ".5rem",
              "paddingLeft": ".1rem",
            }
          }
        />
         E-mail
      </a>
      <a
        data-jsx={3314000957}
      >
        <i
          className="icon icon-instagram"
          data-jsx={1566661193}
          style={
            Object {
              "marginRight": ".5rem",
              "paddingLeft": ".1rem",
            }
          }
        />
         Instagram
      </a>
      <a
        data-jsx={3314000957}
      >
        <i
          className="icon icon-facebook"
          data-jsx={1566661193}
          style={
            Object {
              "marginRight": ".5rem",
              "paddingLeft": ".1rem",
            }
          }
        />
         Facebook
      </a>
    </div>
  </div>
</div>
`;
